components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -432,6 +436,38 @@ public void deleteBotDataEmailList(String uuid) throws APIManagementException, S
         apiMgtDAO.deleteBotDataEmailList(uuid);
     }
 
+    /**
+     * Retrieve all bot detected data
+     *
+     * @return list of bot detected data
+     * @throws APIManagementException
+     */
+    public List<BotDetectionData> retrieveSavedBotData() throws APIManagementException {
```

Comment:
Is 'Saved' required?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/8708/commits/a053882910460fe518c13a17458690efc528ee4f